lib/SILOptimizer/LoopTransforms/ArrayBoundsCheckOpts.cpp

```diff
@@ -177,7 +177,7 @@ static bool isIdentifiedUnderlyingArrayObject(SILValue V) {
   if (isa<SILFunctionArgument>(V))
     return true;
 
-  return false;
+  return lookThroughAddressAndValueProjections(V);
```

Comment:
This is wrong because `lookThroughAddressAndValueProjections` always returns a valid SILValue which is a boolean true. I think you might want to do something like `isIdentifiedUnderlyingArrayObject(lookThroughAddressAndValueProjections)`

Reply:
ughh.. that's what I meant to write. Thanks!